docs(*) auto-generated docs update for 2.1.x with manual changes (#2130)
Kong/docs.konghq.com: app/2.1.x/configuration.md

---

#### port_maps

> With this configuration parameter you can let the Kong to know about the port

Reviewer: ```suggestion
With this configuration parameter, you can let the Kong to know about the port
```

from which the packets are forwarded to it. This is fairly common when running
Kong in a containerized or virtualized environment.

> For example `port_maps=80:8000, 443:8443` instructs Kong that the port 80 is

Reviewer: ```suggestion
For example, `port_maps=80:8000, 443:8443` instructs Kong that the port 80 is
```

> mapped to 8000 (and the port 443 to 8443), where 8000 and 8443 are the ports the

Reviewer: ```suggestion
mapped to 8000 (and the port 443 to 8443), where 8000 and 8443 are the ports that
```

Kong is listening to.

> This parameter helps Kong to set proper forwarded upstream HTTP request header

Reviewer: ```suggestion
This parameter helps Kong to set a proper forwarded Upstream HTTP request header
```

> or to get proper forwarded port with a Kong PDK (in case other means determining

Reviewer: ```suggestion
or to get the proper forwarded port with a Kong PDK (in case other means to determine
```

---

#### ssl_protocols

Enables the specified protocols for client-side connections. The set of
supported protocol versions also depends on the version of OpenSSL Kong was
built with. This value is ignored if `ssl_cipher_suite` is not `custom`.

See http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_protocols

Default: `TLSv1.1 TLSv1.2 TLSv1.3`

---

#### ssl_prefer_server_ciphers

Specifies that server ciphers should be preferred over client ciphers when
using the SSLv3 and TLS protocols. This value is ignored if `ssl_cipher_suite`
is not `custom`.

See
http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_prefer_server_ciphers

Default: `on`

---

#### ssl_session_tickets

> Enables or disables session resumption through TLS session tickets. This is has

Reviewer: ```suggestion
Enables or disables session resumption through TLS session tickets. This has
```

---

#### error_default_type

Default MIME type to use when the request `Accept` header is missing and Nginx
is returning an error for the request.

Accepted values are `text/plain`, `text/html`, `application/json`, and
`application/xml`.

Default: `text/plain`

---

#### upstream_keepalive_pool_size

Sets the default size of the upstream keepalive connection pools.

Upstream keepalive connection pools are segmented by the `dst ip/dst port/SNI`
attributes of a connection.

> A value of `0` will disable upstream keepalive connections by default, forcing

Reviewer: ```suggestion
A value of `0` will disable Upstream keepalive connections by default, forcing
```

> each upstream request to open a new connection.

Reviewer: ```suggestion
each Upstream request to open a new connection.
```

worker process.

The special and default value of `auto` sets this value to `ulimit -n` with the
upper bound limited to 16384 as a measure to protect against excess memory use.

See http://nginx.org/en/docs/ngx_core_module.html#worker_connections

Default: `auto`

---

#### nginx_http_client_header_buffer_size

Sets buffer size for reading the client request headers.

See
http://nginx.org/en/docs/http/ngx_http_core_module.html#client_header_buffer_size

Default: `1k`

---

#### nginx_http_large_client_header_buffers

Sets the maximum number and size of buffers used for reading large clients
requests headers.

See
http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers

Default: `4 8k`

---

#### nginx_http_client_max_body_size

Defines the maximum request body size allowed by requests proxied by Kong,
specified in the Content-Length request header. If a request exceeds this limit,
Kong will respond with a 413 (Request Entity Too Large). Setting this value to 0
disables checking the request body size.

See
http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size

Default: `0`

---

> #### nginx_http_client_body_buffer_size

Reviewer 2: this gained a `nginx_http_` prefix. Is the old one still valid for backwards compatibility?

Defines the buffer size for reading the request body. If the client request
body is larger than this value, the body will be buffered to disk. Note that

> when the body is buffered to disk Kong plugins that access or manipulate the

Reviewer: ```suggestion
when the body is buffered to disk, Kong plugins that access or manipulate the
```

`declarative_config` property, or via the Admin API using the `/config`
endpoint.

When using Postgres as the backend storage, you can optionally enable Kong to
serve read queries from a separate database instance.

> When the number of proxies are large this can greatly reduce the load on the

Reviewer: ```suggestion
When the number of proxies are large, this can greatly reduce the load on the
```

main Postgres instance and achieve better scalability. It may also reduce the
latency jitter if the Kong proxy node's latency to the main Postgres instance is
high.

The read-only Postgres instance only serves read queries and write queries
still goes to the main connection. The read-only Postgres instance can be

> eventual consistent while replicating changes from the main instance.

Reviewer: ```suggestion
eventually consistent while replicating changes from the main instance.
```

At least the `pg_ro_host` config is needed to enable this feature.

> By default all other database config for the read-only connection are inherited

Reviewer: ```suggestion
By default, all other database config for the read-only connection are inherited
```

**pg_ssl_verify** | Toggles server certificate verification if `pg_ssl` is enabled. See the `lua_ssl_trusted_certificate` setting to specify a certificate authority. | `off`
**pg_max_concurrent_queries** | Sets the maximum number of concurrent queries that can be executing at any given time. This limit is enforced per worker process; the total number of concurrent queries for this node will be will be: `pg_max_concurrent_queries * nginx_worker_processes`. The default value of 0 removes this concurrency limitation. | `0`
**pg_semaphore_timeout** | Defines the timeout (in ms) after which PostgreSQL query semaphore resource acquisition attempts will fail. Such failures will generally result in the associated proxy or Admin API request failing with an HTTP 500 status code. Detailed discussion of this behavior is available in the online documentation. | `60000`

> **pg_ro_host** | Same as `pg_host`, but for the read-only connection. Value of `NONE` disables read-only connection. **Note:** please refer to the documentation section above for detailed usage. | `NONE`

Reviewer: ```suggestion
**pg_ro_host** | Same as `pg_host`, but for the read-only connection. Value of `NONE` disables read-only connection. **Note:** Refer to the documentation section above for detailed usage. | `NONE`
```

Time-to-live (in seconds) of an entity from the datastore when cached by this
node.

Database misses (no entity) are also cached according to this setting if you do
not configure `db_cache_neg_ttl`.

> If set to 0 (default), such cached entities or misses never expire

Reviewer: ```suggestion
If set to 0 (default), such cached entities or misses never expire.
```

### Tuning & Behavior section

#### worker_consistency

Defines whether this node should rebuild its state synchronously or
asynchronously (the balancers and the router are rebuilt on updates that affects

> them, e.g. updates to Routes, Services or Upstreams, via the Admin API or

Reviewer: ```suggestion
them; e.g., updates to Routes, Services or Upstreams, via the Admin API or
```
